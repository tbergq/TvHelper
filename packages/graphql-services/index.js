// @flow

export { default as fetch } from './src/fetch';

// graphql shared types
export { default as LoginType } from './src/shared/types/output/LoginType';
export { default as RangeDelete } from './src/shared/types/output/RangeDelete';

// graphql shared resolvers
export { default as LoginResolver, signToken } from './src/shared/resolvers/LoginResolver';

export type { LoggedInUser } from './src/shared/resolvers/LoginResolver';
export type { Apps } from './src/shared/resolvers/LoginResolver';

// others
export { default as connectionFromArray } from './src/connectionFromArray';
export type { ConnectionArguments, Connection } from './src/connectionFromArray';
export type { GraphQLConnectionDefinitions } from './src/types';
