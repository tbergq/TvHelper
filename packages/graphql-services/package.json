{
  "name": "@tbergq/graphql-services",
  "version": "0.0.1",
  "private": true,
  "main": "index.js",
  "dependencies": {
    "@adeira/fetch": "^1.0.0",
    "@kiwicom/graphql-global-id": "^0.13.0",
    "dataloader": "^1.4.0",
    "dotenv": "^8.2.0",
    "graphql": "^14.5.8",
    "graphql-relay": "^0.6.0",
    "jsonwebtoken": "^8.5.1",
    "password-hash": "^1.2.2"
  }
}
