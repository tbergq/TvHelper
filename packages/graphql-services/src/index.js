// @flow

export { default as fetch } from './fetch';

// graphql shared types
export { default as LoginType } from './shared/types/output/LoginType';
export { default as RangeDelete } from './shared/types/output/RangeDelete';

// graphql shared resolvers
export { default as LoginResolver, signToken } from './shared/resolvers/LoginResolver';

export type { LoggedInUser } from './shared/resolvers/LoginResolver';
export type { Apps } from './shared/resolvers/LoginResolver';

// others
export { default as connectionFromArray } from './connectionFromArray';
export type { ConnectionArguments, Connection } from './connectionFromArray';
export type { GraphQLConnectionDefinitions } from './types';
