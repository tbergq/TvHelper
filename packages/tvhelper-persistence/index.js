// @flow

import Favorite from './src/dataObjects/Favorite';

// Repositories
export { default as UserRepository } from './src/repositories/User';
export { default as FavoritesRepository } from './src/repositories/Favorites';
export { default as WatchedEpisodeRepository } from './src/repositories/WatchedEpisode';

// Models
export { default as UserModel } from './src/models/UserModel';
export { default as FavoritesModel } from './src/models/Favorites';

// types
export type FavoriteType = Favorite;

// Connection
export { default as tvHelperConnection } from './src/connection';
