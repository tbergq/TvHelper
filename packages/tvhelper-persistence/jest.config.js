// @flow

const path = require('path');

module.exports = {
  preset: (path.relative(
    __dirname,
    require.resolve('@tbergq/test-preset/node-preset.js'),
  ) /*: string */),
  rootDir: __dirname,
  setupFilesAfterEnv: [(path.join(__dirname, 'scripts', 'setupTest.js') /*: string */)],
};
