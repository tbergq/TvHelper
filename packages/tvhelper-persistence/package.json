{
  "name": "@tbergq/tvhelper-persistence",
  "private": true,
  "version": "0.0.1",
  "main": "index.js",
  "sideEffects": false,
  "dependencies": {
    "@adeira/monorepo-utils": "^0.8.0",
    "dotenv": "^8.2.0",
    "mongodb": "^3.6.0",
    "mongoose": "^5.9.28",
    "password-hash": "^1.2.2"
  },
  "peerDependencies": {
    "mockingoose": "^2.13.2"
  },
  "devDependencies": {
    "mockingoose": "^2.13.2",
    "mongodb-memory-server": "^6.6.3"
  }
}
