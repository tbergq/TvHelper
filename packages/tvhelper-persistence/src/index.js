// @flow

import Favorite from './dataObjects/Favorite';

// Repositories
export { default as UserRepository } from './repositories/User';
export { default as FavoritesRepository } from './repositories/Favorites';
export { default as WatchedEpisodeRepository } from './repositories/WatchedEpisode';

// Models
export { default as UserModel } from './models/UserModel';
export { default as FavoritesModel } from './models/Favorites';

// types
export type FavoriteType = Favorite;

// Connection
export { default as tvHelperConnection } from './connection';
