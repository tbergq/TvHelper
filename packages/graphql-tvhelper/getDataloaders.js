// @flow

import { FavoritesRepository } from '@tbergq/tvhelper-persistence';

import SearchTvShowLoader from './src/search/dataloaders/SearchTvShowLoader';
import UserLoader from './src/account/dataloaders/UserLoader';
import TvDetailLoader from './src/tvshow/dataloaders/TvDetailLoader';
import EpisodesLoader from './src/episode/dataloaders/EpisodesLoader';
import FavoritesLoader from './src/tvshow/dataloaders/FavoritesLoader';

export default function getDataloaders() {
  const favoritesRepository = new FavoritesRepository();
  return {
    searchTvShow: SearchTvShowLoader(),
    user: UserLoader(),
    tvDetail: TvDetailLoader(),
    episodes: EpisodesLoader(),
    favorites: FavoritesLoader(favoritesRepository),
  };
}
