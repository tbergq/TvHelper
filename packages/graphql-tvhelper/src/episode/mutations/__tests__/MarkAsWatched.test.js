// @flow

import { generateExecuteTestQuery } from '@tbergq/graphql-services';
import { UserRepository, tvHelperConnection } from '@tbergq/tvhelper-persistence';

import Mutations from '../../../../TvHelperMutations';
import getDataloaders from '../../../../getDataloaders';

let userId;
const context = {
  user: {},
  dataLoader: {
    tvhelper: getDataloaders(),
  },
};

const mutation = `
mutation {
  markAsWatched(episodeId: "ZXBpc29kZTo2") {
    success
    episode {
      id
    }
  }
}
`;
const executeTestQuery = generateExecuteTestQuery(null, Mutations, context);

describe('MarkAsWatched', () => {
  beforeEach(async () => {
    const user = await UserRepository.createUser({
      password: 'lol',
      username: 'lol',
      email: 'lol',
    });
    userId = user.id;
  });

  afterEach(async () => {
    await tvHelperConnection.collection('users').drop();
    context.user = {};
  });

  it('marks episode as watched', async () => {
    context.user.id = userId;
    const res = await executeTestQuery(mutation);
    expect(res).toMatchSnapshot();
  });

  it('handles not logged in users', async () => {
    const res = await executeTestQuery(mutation);
    expect(res).toMatchSnapshot();
  });
});
