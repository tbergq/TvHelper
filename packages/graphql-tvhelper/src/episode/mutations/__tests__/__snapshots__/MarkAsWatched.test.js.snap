// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkAsWatched handles not logged in users 1`] = `
Object {
  "data": Object {
    "markAsWatched": Object {
      "episode": null,
      "success": false,
    },
  },
}
`;

exports[`MarkAsWatched marks episode as watched 1`] = `
Object {
  "data": Object {
    "markAsWatched": Object {
      "episode": Object {
        "id": "RXBpc29kZTo2",
      },
      "success": true,
    },
  },
}
`;
